flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/local/DependencyTest.java

```diff
@@ -150,4 +154,50 @@ public TestCatalog(String name, String defaultDatabase) {
 			super(name, defaultDatabase);
 		}
 	}
+
+	/**
+	 * A testing catalog for Hivecatalog. In unit test, since there's no Hive metastore
+	 * for the catalog to connect to, we override the open and close methods to not setup any connection.
+	 */
+	public static class TestHiveCatalog extends HiveCatalog {
```

Comment:
The test seems hacky to me. I'm wondering if we can just use embedded hms instead.